### PR TITLE
feat(storefront): BCTHEME-322 add sold-out badges for products on PLP

### PR DESCRIPTION
**Pull request by bc-alexsaiannyi**
#### What?

This PR adds possibility to  setup and show sold-out badges on products before going to PDP. 

#### Tickets / Documentation

- [BCTHEME-322](https://jira.bigcommerce.com/browse/BCTHEME-322)

#### Screenshots (if appropriate)
sold-out-badge_Page_builder
sold-out-badge_in_store
sold-out-badge_homepage

### CODE REVIEW

File: templates/components/products/sales-badge.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{{#if badge_view '===' 'sash'}}
```
Comment (BC-tymurbiedukhin) [inline templates/components/products/sales-badge.html:1]: I believe product-badge name suites better for this component

File: schemaTranslations.json [hunk 2]
```diff
@@ -1076,6 +1076,13 @@
     "uk": "Показати значки продажу товарів",
     "zh": "显示产品降价标志"
   },
+  "i18n.ShowProductSoldOutBadges": {
+    "default": "Show product sold-out badges",
+    "fr": "Show product sold-out badges",
+    "it": "Show product sold-out badges",
+    "uk": "Показати значки розпроданих товарів",
```
Comment (BC-tymurbiedukhin) [inline schemaTranslations.json:1083]: Етикетки чи значки?

File: assets/scss/layouts/products/_productSaleBadges.scss [hunk 3]
```diff
@@ -22,11 +26,20 @@
     z-index: zIndex("lower");
 }
 
-.sale-flag-star:before {
+.sold-out-flag-star,
+.sold-out-flag-star:before,
+.sold-out-flag-star:after {
+    @extend .sale-flag-star;
```
Comment (BC-tymurbiedukhin) [inline assets/scss/layouts/products/_productSaleBadges.scss:32]: Could we make a class base class for start and reuse it for sale and sold out classes?

File: templates/components/products/sales-badge.html (hunk 1 above)
Comment (bc-alexsaiannyi) [reply templates/components/products/sales-badge.html]: I don't mind. changed

File: schemaTranslations.json (hunk 2 above)
Comment (bc-alexsaiannyi) [reply schemaTranslations.json:1083]: вибрав значки,  щоб було співставно з "Показати значки продажу товарів". Можна замінити, але тоді це робити в обох варіантах.

File: assets/scss/layouts/products/_productSaleBadges.scss (hunk 3 above)
Comment (bc-alexsaiannyi) [reply assets/scss/layouts/products/_productSaleBadges.scss]: I added placeholder selector that is used as base class for sales and sold-out classes

**Review by BC-tymurbiedukhin — APPROVED** [review]
LGTM

**Pull request merged**